lib/measured/measurable.rb

```diff
@@ -4,26 +4,10 @@ class Measured::Measurable < Numeric
   attr_reader :unit, :value
 
   def initialize(value, unit)
-    raise Measured::UnitError, "Unit cannot be blank" if unit.blank?
-    raise Measured::UnitError, "Unit #{ unit } does not exist" unless self.class.conversion.unit_or_alias?(unit)
-
-    @value = case value
-    when Float
-      BigDecimal(value, Float::DIG+1)
-    when Integer
-      BigDecimal(value)
-    when NilClass
-      raise Measured::UnitError, "Unit value cannot be nil"
-    when BigDecimal
-      value
-    else
-      if value.blank?
-        raise Measured::UnitError, "Unit value cannot be blank"
-      else
-        BigDecimal(value)
-      end
-    end
+    raise Measured::UnitError, "Unit '#{unit}' does not exist" unless self.class.conversion.unit_or_alias?(unit)
+    raise Measured::UnitError, "Unit value cannot be blank" if value.blank?
 
+    @value = BigDecimal(value, value.is_a?(Float) ? Float::DIG + 1 : 0)
```

Comment:
> Also, this works if value is a BigDecimal object?

Yep, but I just realized this will actually clone the object, which is a bigger performance loss than anticipated. I'm going to revert to the case statement, but pull the blank value check out of it.